test/src/test/java/org/corfudb/runtime/LogReplicationServiceClientTest.java

```diff
@@ -0,0 +1,380 @@
+        final Table<ClientDestinationInfoKey, DestinationInfoVal, SampleSchema.ManagedResources> table = store.openTable(
+                namespace,
+                metadataTableName,
+                ClientDestinationInfoKey.class,
+                DestinationInfoVal.class,
+                SampleSchema.ManagedResources.class,
+                TableOptions.fromProtoSchema(SampleSchema.Uuid.class));
+
+        LogReplicationServiceClient client = new LogReplicationServiceClient(runtime, clientName);
+
+        // Test adding destinations to a logicalGroup
+        List<String> destinationsToAdd = Arrays.asList("DESTINATION1", "DESTINATION2");
+        Assert.assertTrue(client.addDestination("LOGICALGROUP", destinationsToAdd));
+        List<String> destinationsToAdd2 = Arrays.asList("DESTINATION3", "DESTINATION4");
+        Assert.assertTrue(client.addDestination("LOGICALGROUP", destinationsToAdd2));
+
+        // Test adding existing destinations
+        List<String> destinationsToAdd3 = Arrays.asList("DESTINATION1", "DESTINATION4");
+        Assert.assertTrue(client.addDestination("LOGICALGROUP", destinationsToAdd3));
+
+        // Test adding with request having a malformed logicalGroup
+        List<String> destinationsToAdd4 = Collections.singletonList("DESTINATION1");
+        Assert.assertFalse(client.addDestination("", destinationsToAdd4));
+        Assert.assertFalse(client.addDestination(null, destinationsToAdd4));
+
+        // Test adding with request having a malformed destination
+        List<String> destinationsToAdd5 = Collections.singletonList("");
+        List<String> destinationsToAdd6 = Collections.singletonList(null);
+        List<String> destinationsToAdd7 = Arrays.asList("DESTINATION1", "DESTINATION1");
+        Assert.assertFalse(client.addDestination("LOGICALGROUP", destinationsToAdd5));
+        Assert.assertFalse(client.addDestination("LOGICALGROUP", destinationsToAdd6));
+        Assert.assertFalse(client.addDestination("LOGICALGROUP", destinationsToAdd7));
+        Assert.assertFalse(client.addDestination("LOGICALGROUP", (List<String>) null));
+
+        // Test adding destinations to new logicalGroup
+        List<String> destinationsToAdd8 = Arrays.asList("DESTINATION1", "DESTINATION2");
+        Assert.assertTrue(client.addDestination("LOGICALGROUP2", destinationsToAdd8));
+
+        // Test addition after removal in new logicalGroup
+        Assert.assertTrue(client.removeDestination("LOGICALGROUP2", "DESTINATION2"));
+        List<String> destinationsToAdd9 = Arrays.asList("DESTINATION3", "DESTINATION4");
+        Assert.assertTrue(client.addDestination("LOGICALGROUP2", destinationsToAdd9));
+
+        // Test table size is correct
+        Assert.assertEquals(2, table.count());
+
+        final int batchSize = 2;
+        final HashMap<String, Integer> numDestinations = new HashMap<>();
+        numDestinations.put("LOGICALGROUP", 4);
+        numDestinations.put("LOGICALGROUP2", 3);
```

Comment:
I think the magic numbers here prevent the project from building, both locally and on Github Action env

Reply:
I think I have more instances elsewhere in the test code too, will work on getting that cleaned up.